### ISSUE TITLE
Platform-level non-unique biport identifiers

### ISSUE THREAD

**Issue by jeffesquivels** [HPENetworking/topology_docker#44]
On [line 130](https://github.com/HPENetworking/topology_docker/blob/863d5773c641c8f1f72e9c30fb62269779b4fecb/lib/topology_docker/platform.py#L130) of platform.py, we use the biport's identifier as the key to add it to the platform's nmlbiport_iface_map:

```python
self.nmlbiport_iface_map[biport.identifier] = {
```

Which means that a newer biport with the same identifier as one that's been already added to the platform will effectively replace it in the nmlbiport_iface_map. Because of this, one would think that having non-unique biport identifiers on the same platform shouldn't be supported, but on [lines 185 through 193](https://github.com/HPENetworking/topology_docker/blob/863d5773c641c8f1f72e9c30fb62269779b4fecb/test/test_topology_docker_platform.py#L185-L193) of test/test_topology_docker_platform.py we are defining two biports with identifiers '3' and '4', adding them to the platform, and then defining two different biports with those same identifiers and adding them to the platform too:

```pyhton
s1p1 = BidirectionalPort(identifier='3')
s1p2 = BidirectionalPort(identifier='4')
platform.add_biport(s1, s1p1)
platform.add_biport(s1, s1p2)

s2p1 = BidirectionalPort(identifier='3')
s2p2 = BidirectionalPort(identifier='4')
platform.add_biport(s2, s2p1)
platform.add_biport(s2, s2p2)
```

Which would make one think that this is indeed a supported scenario, so, are non-unique biport identifiers on the same platform supported? If yes, platform.py should be fixed, if no, the test should be fixed and we should probably add some kind of error if someone tries to add an already existing biport identifier to the platform.

**Comment by ocelotl** [comment]
Looks like a bug to me, I think we should use both node and biport as keys for that dictionary.

**Comment by ocelotl** [comment]
The usage of ``add_biport`` mentioned in this issue never happens when the Topology calls that function with a biport created like this:

 https://github.com/HPENetworking/topology/blob/master/lib/topology/manager.py#L152-L155

So, I'm closing this one since ths test-case issue was already fixed in b93f31b.

P.D.

Some think we should check for key uniqueness in the ``add_biport`` method. If so, please open a new issue to track that enhancement.
